applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.h

```diff
@@ -0,0 +1,73 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Bodhinanda Chandra
+//
+
+
+#ifndef KRATOS_MPM_PARTICLE_GENERATOR_UTILITY
+#define KRATOS_MPM_PARTICLE_GENERATOR_UTILITY
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "particle_mechanics_application_variables.h"
+
+
+namespace Kratos
+{
+namespace MPMParticleGeneratorUtility
+{
+
+    typedef std::size_t IndexType;
+
+    typedef std::size_t SizeType;
+
+    /**
+     * @brief Function that return matrix of shape function value for 16 particles.
+     * @details It is only possible to be used in 2D Triangular.
+     */
+    Matrix MP16ShapeFunctions();
```

Comment:
Actually, you could make a shapefuntiongenerator utility. Anyways, shouldn't be needed for long.
.